homeassistant/components/bluetooth/update_coordinator.py

```diff
@@ -53,10 +54,37 @@ def _async_cancel() -> None:
 
         return _async_cancel
 
+    @callback
+    @abstractmethod
+    def _async_handle_bluetooth_event(
+        self,
+        service_info: BluetoothServiceInfoBleak,
+        change: BluetoothChange,
+    ) -> None:
+        """Handle a bluetooth event."""
+
+    @property
+    def name(self) -> str:
+        """Return last known name of the device."""
+        if service_info := async_last_service_info(
+            self.hass, self.address, self.connectable
+        ):
+            return service_info.name
+        return self._last_name
+
+    @property
+    def last_seen(self) -> float:
+        """Return the last time the device was seen."""
+        if service_info := async_last_service_info(
+            self.hass, self.address, self.connectable
+        ):
+            return service_info.time
+        return self._last_unavailable
```

Comment:
a device can be seen many times after it has been last_unavailable. Is it correct to return this here?

Reply:
Last seen should be the last time we got an advertisement for the device. I'm pretty sure the logic is right but will definitely add a test to validate it